code/modules/reagents/chemistry/holder.dm

```diff
@@ -696,3 +696,12 @@ var/const/INJECT = 5 //injection
 		qdel(reagents)
 	reagents = new/datum/reagents(max_vol)
 	reagents.my_atom = src
+
+/proc/get_random_reagent_id()	// Returns a random reagent ID minus blacklisted reagents
+	var/list/random_reagents = list()
```

Comment:
reagents aren't created midround, so this list should be static (yes, still within the proc!) to save doing the looping every time.

(just make sure to put the loop under a ```if(!random_reagents.len)``` check)